pallets/exchange/src/lib.rs

```diff
@@ -287,6 +267,61 @@ decl_module! {
 
 // "Internal" functions, callable by code.
 impl<T: Config> Module<T> {
+	/// Register SELL or BUY intention
+	fn register_intention(
+		who: &T::AccountId,
+		intention_type: IntentionType,
+		assets: AssetPair,
+		amount_in: Balance,
+		amount_out: Balance,
+		limit: Balance,
+		discount: bool,
+	) -> dispatch::DispatchResult {
+		let intention_count = ExchangeAssetsIntentionCount::get(assets.ordered_pair());
+
+		let intention_id = Self::generate_intention_id(who, intention_count, &assets);
+
+		let intention = Intention::<T> {
+			who: who.clone(),
+			assets,
+			amount_sell: amount_in,
```

Comment:
do we want to unify these to in/out?

Reply:
If we do we should find every use here and change it

Reply:
Yes, I agree. will do.